concordia/views.py

```diff
@@ -1128,17 +1129,34 @@ def reserve_asset(request, *, asset_pk):
 
     timestamp = now()
 
-    # First clear old reservations, with a grace period:
+    # First clear old reservations, with a grace period.
     cutoff = timestamp - (
         timedelta(seconds=2 * settings.TRANSCRIPTION_RESERVATION_SECONDS)
     )
 
+    reservations = []
+
     with connection.cursor() as cursor:
-        cursor.execute(
-            "DELETE FROM concordia_assettranscriptionreservation WHERE updated_on < %s",
+        rows_to_release = cursor.execute(
+            (
+                "DELETE FROM concordia_assettranscriptionreservation "
+                "WHERE updated_on < %s "
+                "RETURNING asset_id AS asset_pk, user_id AS user_pk"
+            ),
             [cutoff],
         )
 
+    if rows_to_release:
```

Comment:
I would have thought that this block would need to be inside the cursor context manager to avoid the cursor being closed before the results are retrieved

Reply:
It passed the unit tests... we should definitely test it more.

Reply:
I tweaked it in the main prototype branch